tests/src/native_vp/pos.rs

```diff
@@ -401,7 +401,10 @@ mod tests {
                         address,
                         consensus_key,
                         commission_rate: _,
+                        max_commission_rate_change: _,
                     } => {
+                        // TODO: should there be preconditions for commission
+                        // rates here?
```

Comment:
I don't think we need any - a new validator can choose any rates in `0..=1` and we're only generating values in this range. 

We can however add a `Transition` for `ValidPosAction::ChangeCommissionRate`, for which the model would need to have a pre-condition that ensures that the new rate is less than or equal to the validator's `max_commission_rate_change` with respect to its last epoch's rate. I think this can be done in in the abstract SM by e.g. looking through `self.all_valid_actions().into_iter().rev()` for the last rate value (either from a previous `ValidPosAction::ChangeCommissionRate` or `ValidPosAction::InitValidator` if it's not been changed before) before the last epoch switch. However, the epoch switch is currently not preserved in the abstract state machine - we only have the current `epoch`, so we might need to change that first.